File: 프로그래머스/2/12936. 줄 서는 방법/줄 서는 방법.java
import java.util.*;
class Solution {
    public int[] solution(int n, long k) {
        int[] answer = new int[n];
        List<Integer> arr = new ArrayList<>();
        long total = 1;
        for (int i = 1; i <= n; i++) {
            arr.add(i);
            total *= i;
        }
        
        k--; // 인덱스 0부터니까
        for (int i = 0; i < n; i++) {
            total /= (n-i);
            int idx = (int) (k/total);
            answer[i] = arr.get(idx);
            arr.remove(idx);
            k %= total;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12936. 줄 서는 방법/줄 서는 방법.py
from collections import deque
def factorial(n):
    if n == 1:
        return 1
    return n * factorial(n-1)

def solution(n, k):
    answer = []
    queue = deque([i for i in range(1, n+1)])
    
    while n > 1:
        fac = factorial(n-1)
        num = queue[(k-1)//fac]
        answer.append(num)
        queue.remove(num)
        n -= 1
        k %= fac
    answer.append(queue[-1])
    return answer